discoverd/server/handler.go

```diff
@@ -22,6 +24,16 @@ import (
 // StreamBufferSize is the size of the channel buffer used for event subscription.
 const StreamBufferSize = 64 // TODO: Figure out how big this buffer should be.
 
+func loggerFn(handler http.Handler, logger log.Logger, clientIP string, rw *hh.ResponseWriter, req *http.Request) {
+	start := time.Now()
+	handler.ServeHTTP(rw, req)
+	switch rw.Status() {
+	case 200, 307, 0: // 0 == 200 OK
+	default:
+		logger.Info("request completed", "status", rw.Status(), "duration", time.Since(start))
```

Comment:
Since this doesn't log the request start, it should also log th request method/path/client ip.